src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.StackRowStack.cs

```diff
@@ -23,8 +23,8 @@ internal StackRowStack(int initialSize)
 
             public void Dispose()
             {
-                byte[] toReturn = _rentedBuffer;
-                _rentedBuffer = null;
+                byte[]? toReturn = _rentedBuffer;
+                _rentedBuffer = null!;
```

Comment:
Make `_rentedBuffer` nullable and add `Debug.Assert(_rentedBuffer != null);` on `Pop` and `Enlarge` instead.

Reply:
This is same as https://github.com/dotnet/runtime/pull/114#discussion_r354091014